src/plonkish/ir/sc.rs

```diff
@@ -7,11 +7,20 @@ use super::{
     Circuit,
 };
 
+#[derive(Debug)]
 pub struct SuperCircuit<F, MappingArgs> {
     sub_circuits: Vec<Circuit<F>>,
     mapping: MappingGenerator<F, MappingArgs>,
 }
 
+impl<F: Debug, MappingArgs: Debug> Debug for MappingGenerator<F, MappingArgs> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("MappingGenerator")
```

Comment:
@rutefig why is the default Debug not good? it will not print the pointer, right?

Reply:
Yes, it does not print the pointer

Reply:
But if we don't print the pointer we still need to implement Debug right? we cannot use the default one because the pointer does not implements Debug

Reply:
I think you can use the default one, it just ignore the pointer

Reply:
It complaints if I try to use the default and it doesn't allow to use it